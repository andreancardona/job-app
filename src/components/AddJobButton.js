import React from 'react';
import { useHistory } from 'react-router-dom';
import '../styles/_buttons.scss'

const AddJobButton = (props) => {
  
  let history = useHistory();

  return (
    <button className="add-save-button" onClick={() => {props.setAdd() || history.push("/job-form")}}>
      Add job
    </button>
  )
}

export default AddJobButton;