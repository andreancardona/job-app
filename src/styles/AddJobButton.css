.add-job-button {
  color: #ffff;
  font-family: Helvetica Neue;
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 18px;
  background: #085FF7;
  padding: 13px, 16px, 13px, 16px;
  border-radius: 8px;
  width: 91px;
  height: 44px;
  border: none;
}