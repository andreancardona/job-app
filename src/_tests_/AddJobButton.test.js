import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import AddJobButton from '../components/AddJobButton';

describe('The <AddJobButton /> component', () => {
  test('AddJobButton renders with AddJobButton props', () => {
    const tree = renderer.create(<AddJobButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('AddJobButton should not fail any accessibility tests', async () => {
    const { container } = render(<AddJobButton />);
    expect(await axe(container)).toHaveNoViolations();
  })
});
