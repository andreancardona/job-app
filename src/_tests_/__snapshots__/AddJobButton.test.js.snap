// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <AddJobButton /> component AddJobButton renders with AddJobButton props 1`] = `
<button
  className="add-save-button"
  onClick={[Function]}
>
  Add job
</button>
`;
